### ISSUE TITLE
Playing through a USB-C to 3.5mm adapter introduces distortion

### ISSUE THREAD

**Issue by gravelld** [daneren2005/Subsonic#1079]
It appears that sometimes when listening to audio emitted from D-Sub via a USB-C to 3.5mm adapter distortion is heard. This is not observed in other audio apps. It happens at different times - not necessarily the start/end of a track.

AFAIK D-Sub just passes the audio through "standard" Android audio APIs, right? I've read some things about custom USB drivers being used by some audio apps to improve performance.

Has anyone else observed this - is it maybe related to processor load?

**Comment by daneren2005** [comment]
You are right that DSub just uses the standard audio API.  As time has gone on basically every other audio app has had to write their own audio stack because Android's is shit.  And as more people have stopped using the standard Android API, device manufacturers have stopped giving af, making it even worse.  Unfortunately, I'm really not looking at introducing my own audio stack at this time.

**Comment by gravelld** [comment]
Thanks for the reply. Do alternative audio stack libraries exist that could be used?

You can close this as "won't fix" if you like... I don't want to dictate how you manage issues/enhancements such as these, so I'll leave it up to you.

**Comment by Efreak** [comment]
You could try using BubbleUPnP for output, and control it via DLNA. Not sure if it would work, though. You could also try using something like Poweramp equalizer; I know Poweramp has good audio output, but [not sure if that's something the equalizer can use](https://forum.powerampapp.com/topic/22116-hi-res-output-and-volume-over-100/?do=findComment&comment=101341).
